python/ray/data/dataset.py

```diff
@@ -2267,22 +2295,52 @@ def to_tf(
         except ImportError:
             raise ValueError("tensorflow must be installed!")
 
+        # `output_signature` can be a tuple but not a list. See
+        # https://stackoverflow.com/questions/59092423/what-is-a-nested-structure-in-tensorflow.
+        if isinstance(output_signature, list):
+            output_signature = tuple(output_signature)
+
         def make_generator():
             for batch in self.iter_batches(
                 prefetch_blocks=prefetch_blocks,
                 batch_size=batch_size,
                 batch_format="pandas",
             ):
                 if label_column:
-                    target_col = batch.pop(label_column)
-                if feature_columns:
-                    batch = batch[feature_columns]
+                    targets = batch.pop(label_column).values
+
+                features = None
+                if feature_columns is None:
+                    features = batch.values
+                elif isinstance(feature_columns, list):
+                    if all(isinstance(element, str) for element in feature_columns):
+                        features = batch[feature_columns].values
+                    elif all(isinstance(element, list) for element in feature_columns):
```

Comment:
Yeah, wasn't really sure what to name this since each element of `feature_columns` is a list of feature columns. In any case, I renamed it `element` as `columns`